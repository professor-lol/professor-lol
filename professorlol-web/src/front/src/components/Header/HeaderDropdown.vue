<template>
  <b-nav-item-dropdown :right="right" :no-caret="noCaret">
    <template slot="button-content">
      <slot name="header">
        &#10068;
      </slot>
    </template>
    <slot name="dropdown">
      <div :style="{ right: 'auto', height: '200px' }">
        <span class="text-center">dropdown</span>
      </div>
    </slot>
  </b-nav-item-dropdown>
</template>
<script>
export default {
  name: 'HeaderDropdown',
  props: {
    right: {
      type: Boolean,
      default: false
    },
    noCaret: {
      type: Boolean,
      default: false
    }
  }
}
</script>
