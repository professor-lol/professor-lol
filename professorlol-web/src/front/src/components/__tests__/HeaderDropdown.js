import { shallowMount } from '@vue/test-utils'
import HeaderDropdown from '../Header/HeaderDropdown';

describe("HeaderDropdown.vue", () => {
  // Inspect the raw component options
  it("should have default props", () => {
    const wrapper = shallowMount(HeaderDropdown, {
      propsData: {
        right: false,
        noCaret: false
      }
    })
    expect(wrapper.props().right).toBe(false)
    expect(wrapper.props().noCaret).toBe(false)
  });
  it('renders correctly', () => {
    const wrapper = shallowMount(HeaderDropdown)
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.find('span').text()).toBe('dropdown')
    expect(wrapper.find('span').classes()).toContain('text-center')
  })
});
