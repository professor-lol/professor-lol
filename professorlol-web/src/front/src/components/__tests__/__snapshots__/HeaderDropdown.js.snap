// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderDropdown.vue renders correctly 1`] = `
<b-nav-item-dropdown>
  <template>
    
      ‚ùî
    
  </template>
   
  <div
    style="height: 200px;"
  >
    <span
      class="text-center"
    >
      dropdown
    </span>
  </div>
</b-nav-item-dropdown>
`;
